### PR TITLE
Minio support to substitute s3 on dev

### PR DESCRIPTION
**Pull request by dLobatog**
This commit adds a minio.io container to docker-compose. This container
has the same API as s3 for writing/reading/etc, and it also runs on port
9000 of the Docker host.

Therefore this allows Insights app developers to use insights-upload
with this S3-compatible backend, instead of using S3 accounts (which
cost $$$), or localhost - which doesn't work for developing insights
apps as the consumers will need to fetch the file from a S3 bucket, not
from a file in a container.

I think it would be possible to create the buckets automatically with
docker compose, as the buckets are automatically created if a new
directory is created in /mnt/data, but I don't know how exactly can I do
that: e.g:  /mnt/data/insights-upload-perm-test/ creates a bucket

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -5,12 +5,24 @@ services:
     environment:
       - ZOOKEEPER_CLIENT_PORT=32181
       - ZOOKEEPER_SERVER_ID=1
+  minio:
+    image: minio/minio
+    command: server /data
+    volumes:
+      - '/mnt/data:/data'
+      - '/mnt/config:/root/.minio'
```
Comment (SteveHNH) [inline docker/docker-compose.yml]: Getting these mounts to work requires disabling SELinux as the host mount points do not have the proper security context. We should probably come up with a way to set the directories to use the proper context prior to starting up.

Not sure if we can do this within docker-compose, but we can at least make a note in the README about it. 
`chcon -R -u system_u -t container_file_t /mnt/data`
`chcon -R -u system_u -t container_file_t /mnt/config`

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -27,8 +39,9 @@ services:
     ports:
       - 8080:8888
     environment:
-      - AWS_ACCESS_KEY_ID
-      - AWS_SECRET_ACCESS_KEY
+      - AWS_ACCESS_KEY_ID=BQA2GEXO711FVBVXDWKM
```
Comment (eduardocerqueira) [inline docker/docker-compose.yml]: Setting the variables here in the docker-compose we are losing the ability to override it dynamically. How about if we just load the variable here -- as it was before -- and the value set will be outside of docker-compose? it applies for 
```
AWS_*
S3_ENDPOINT_URL
```
variables.

**Comment by Glutexo** [comment]
@dLobatog Any plans on going on with this? I don’t actually need it, just saw it here and remembered that I did a review, so just asking.

**Comment by dLobatog** [comment]
@Glutexo Yup, sorry it's taking a while - I've been focused on compliance for a while but I will try to fix this soon.

File: docker/docker-compose.yml (hunk 1 above)
Comment (akofink) [reply docker/docker-compose.yml]: It seems to work as expected with SELinux with `:Z`

```yaml
      - /mnt/data:/data:Z
      - /mnt/config:/root/.minio:Z
```

**Comment by akofink** [comment]
Let's get this unblocked. It seems we need to update to use env variables for mounted volumes, fix SELinux with `:Z`, and ensure kafka actually depends on minio (afaict, it doesn't, and we can remove the depends_on).

**Comment by dLobatog** [comment]
Something odd happened with this PR - apparently when this repo went public, the fork link was removed so no matter if I updated the dlobatog/insights-upload 'minio_support' branch, the PR would never update. I can't reopen the PR even now because of that, I had to fork again at https://github.com/dlobatog/insights-upload-1 and create a new branch.

**Pull request closed without merging**